### ISSUE TITLE
RFC: New capability `wdio:capabilityName` that takes precedence in lib/helpers/sanitize.caps

### ISSUE THREAD

**Issue by spencerwilson-optimizely** [webdriverio/webdriverio#2360]
## Motivation

tl;dr [`sanitize.caps`](https://github.com/webdriverio/webdriverio/blob/76186ed3d6a468a36826d17c5e369b69a5491afa/lib/helpers/sanitize.js#L21-L43) can alias several capabilities objects to the same `sanitizedCapabilityName`, which can cause problems downstream in a reporter like wdio-junit-reporter.

Neither the [spec](https://www.w3.org/TR/webdriver/#capabilities) (admittedly, in Candidate Recommendation stage) nor [BrowserStack](https://www.browserstack.com/automate/capabilities) recognize a `deviceName` capability, which is what webdriverio consults during sanitization. Rather, BrowserStack has an [extension capability](https://www.w3.org/TR/webdriver/#dfn-extension-capabilities) `device` that is semantically the same as a `deviceName` (the fact that it should have been called `browserstack:device` per the spec is another issue).

As a result, as a BrowserStack customer, different versions of mobile devices can be aliased to the same string after passing through `sanitize.caps`. This becomes the first segment of the `classname` in wdio-junit-reporter ([src](https://github.com/webdriverio/wdio-junit-reporter/blob/master/lib/reporter.js#L47)), and so the fact that the tests originated from different capabilities is erased.

## Ideas for solution

- Change references to `caps.deviceName` to `caps.deviceName || caps.device`. I think that might resolve the one failure case I've identified (mobile devices).
- Support a `wdio:capabilityName` capability. If defined, then `sanitizeString(caps['wdio:capabilityName'])` (or `wdio:configName`, `wdio:browserName`, etc.) would be the return value of `sanitize.caps`.

If the maintainers (by the way, thank you!) agree this is a problem, I'm happy to submit a PR implementing the desired resolution. Cheers.

**Comment by christian-bromann** [comment]
This should be handled in the reporter. So more or less solution 1 would work.

**Comment by christian-bromann** [comment]
This issue was moved to webdriverio/wdio-junit-reporter#78
